### DISCUSSION TITLE
How would you build a custom environment?

### DISCUSSION THREAD

**Discussion by 4eshanb** [PKU-Alignment/safety-gymnasium#90] [General]
Hi, 
I'm trying to build a custom environment with this library. I can see how you define a task:
https://www.safety-gymnasium.com/en/latest/components_of_environments/tasks/task_example.html
https://github.com/PKU-Alignment/safety-gymnasium/tree/main#customize-your-environments

But, how would you call these custom environments?
Would you still use the make command?

**Comment by muchvo** [comment]
That's a great question. We have inherited the Gymnasium API, so you can refer to [this code](https://github.com/PKU-Alignment/safety-gymnasium/blob/main/safety_gymnasium/__init__.py) for guidance on how to register your environment.
